### PR TITLE
Restaurant model

### PR DESCRIPTION
**Pull request by eyalse**
1. Created restaurant model
2. Added Rspec env
3. Created factory for restaurant
4. Created test for unique model

### CODE REVIEW

File: app/models/restaurant.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Restaurant < ApplicationRecord
+  validates :name, :cuisine, :address, presence: true
+  validates_inclusion_of :accepts_tenbis, in: [true, false]
```
Comment (pazaricha) [inline app/models/restaurant.rb]: Nice

File: app/models/restaurant.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  validates :name, uniqueness: true
+  validates :rating, numericality: { greater_than_or_equal_to: 0.01, less_than_or_equal_to: 3 }
```
Comment (pazaricha) [inline app/models/restaurant.rb]: Initially, I believe the rating would be 0, so perhaps change it to  `greater_than_or_equal_to: 0` ?

File: db/migrate/20180530131654_add_rating_to_restaurants.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddRatingToRestaurants < ActiveRecord::Migration[5.2]
+  def change
+    add_column :restaurants, :rating, :decimal
```
Comment (pazaricha) [inline db/migrate/20180530131654_add_rating_to_restaurants.rb]: perhaps add a default value of 0 ?

File: spec/models/restaurant_spec.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Restaurant, type: :model do
+  describe 'restaurant with valid input' do
+    let(:rest) { FactoryBot.create(:restaurant, :asian) }
+    it 'should throw error for unique name' do
```
Comment (pazaricha) [inline spec/models/restaurant_spec.rb]: It's usually an overkill to test rails validation as they are already tested on the framework level but it's great you've added a test for the sake of practice.

Great spec!

File: spec/models/restaurant_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Restaurant, type: :model do
+  describe 'restaurant with valid input' do
+    let(:rest) { FactoryBot.create(:restaurant, :asian) }
```
Comment (pazaricha) [inline spec/models/restaurant_spec.rb]: Personally, I prefer clarity over saving a couple of chars when it comes to naming things (variables, methods names etc). So I would change it to:
`let(:restaurant)`

But it's up to you.

Comment (eyalse) [reply spec/models/restaurant_spec.rb]: Gotcha

File: spec/models/restaurant_spec.rb (hunk 4 above)
Comment (eyalse) [reply spec/models/restaurant_spec.rb]: Thanks

File: db/migrate/20180530131654_add_rating_to_restaurants.rb (hunk 3 above)
Comment (eyalse) [reply db/migrate/20180530131654_add_rating_to_restaurants.rb]: Yes.

File: app/models/restaurant.rb (hunk 2 above)
Comment (eyalse) [reply app/models/restaurant.rb]: right.. i was thinking about a `price` but this is clearly another case.

Thanks! will update.

File: app/models/restaurant.rb (hunk 1 above)
Comment (eyalse) [reply app/models/restaurant.rb]: Thanks

**Pull request merged**